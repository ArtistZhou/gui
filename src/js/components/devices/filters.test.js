import React from 'react';
import renderer from 'react-test-renderer';
import Filters from './filters';

describe('Filters Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Filters attributes={[{ key: 'testkey', value: 'testvalue' }]} filters={[]} onFilterChange={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
