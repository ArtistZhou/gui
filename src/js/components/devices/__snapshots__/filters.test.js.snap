// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <button
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-textSecondary-5 MuiButton-flat-6 MuiButton-flatSecondary-8"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100,
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-30 buttonLabelIcon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      Filters
    </span>
    <span
      className="MuiTouchRipple-root-53"
    />
  </button>
</div>
`;
